protocol/v2/ssv/validator/non_committee_validator.go

```diff
@@ -106,3 +155,125 @@ func nonCommitteeInstanceContainerCapacity(fullNode bool) int {
+		if uint64(len(rootSignatures)) >= validator.Quorum() {
+			key := validatorIndexAndRoot{msg.ValidatorIndex, msg.SigningRoot}
+			longestSigners := quorums[key]
+			if newLength := len(rootSignatures); newLength > len(longestSigners) {
+				newSigners := make([]spectypes.OperatorID, 0, newLength)
+				for signer := range rootSignatures {
+					newSigners = append(newSigners, signer)
+				}
+				slices.Sort(newSigners)
+				quorums[key] = newSigners
+			}
+		}
+	}
+	return quorums, nil
+}
+
+// Stores the container's existing signature or the new one, depending on their validity. If both are invalid, remove the existing one
+// copied from BaseRunner
+func (ncv *CommitteeObserver) resolveDuplicateSignature(container *specssv.PartialSigContainer, msg *spectypes.PartialSignatureMessage, share *types.SSVShare) {
+	// Check previous signature validity
+	previousSignature, err := container.GetSignature(msg.ValidatorIndex, msg.Signer, msg.SigningRoot)
+	if err == nil {
+		err = ncv.verifyBeaconPartialSignature(msg.Signer, previousSignature, msg.SigningRoot, share)
+		if err == nil {
+			// Keep the previous sigature since it's correct
+			return
+		}
+	}
+
+	// Previous signature is incorrect or doesn't exist
+	container.Remove(msg.ValidatorIndex, msg.Signer, msg.SigningRoot)
+
+	// Hold the new signature, if correct
+	err = ncv.verifyBeaconPartialSignature(msg.Signer, msg.PartialSignature, msg.SigningRoot, share)
+	if err == nil {
+		container.AddSignature(msg)
+	}
+}
+
+// copied from BaseRunner
+func (ncv *CommitteeObserver) verifyBeaconPartialSignature(signer uint64, signature spectypes.Signature, root [32]byte, share *types.SSVShare) error {
+	types.MetricsSignaturesVerifications.WithLabelValues().Inc()
+
+	for _, n := range share.Committee {
+		if n.Signer == signer {
+			pk, err := types.DeserializeBLSPublicKey(n.SharePubKey)
+			if err != nil {
+				return fmt.Errorf("could not deserialized pk: %w", err)
+			}
+
+			sig := &bls.Sign{}
+			if err := sig.Deserialize(signature); err != nil {
+				return fmt.Errorf("could not deserialized Signature: %w", err)
+			}
+
+			if !sig.VerifyByte(&pk, root[:]) {
+				return fmt.Errorf("wrong signature")
+			}
+			return nil
+		}
+	}
+	return fmt.Errorf("unknown signer")
+}
+
+func (ncv *CommitteeObserver) OnProposalMsg(msg *queue.DecodedSSVMessage) {
+	mssg := &specqbft.Message{}
+	if err := mssg.Decode(msg.SSVMessage.GetData()); err != nil {
+		ncv.logger.Debug("❗ failed to get decode ssv message", zap.Error(err))
+		return
+	}
+
+	// decode consensus data
+	beaconVote := &spectypes.BeaconVote{}
+	if err := beaconVote.Decode(msg.SignedSSVMessage.FullData); err != nil {
+		ncv.logger.Debug("❗ failed to get beacon vote data", zap.Error(err))
+		return
+	}
+	cid := spectypes.CommitteeID(msg.GetID().GetDutyExecutorID()[16:])
+
+	ncv.logger.Info("✅ Got proposal message", fields.CommitteeID(cid))
+	ncv.Roots[beaconVote.BlockRoot] = spectypes.BNRoleSyncCommittee
```

Comment:
how come its always sync committee?

Reply:
@y0sher because if the partialSig root is equal to beaconVote root it means it sync_committee